WalletWasabi/WabiSabi/Backend/WebClients/BlockstreamApiClient.cs

```diff
@@ -0,0 +1,57 @@
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Nito.AsyncEx;
+using WalletWasabi.Tor.Http.Extensions;
+using WalletWasabi.WabiSabi.Backend.Statistics;
+using WalletWasabi.WabiSabi.Backend.WebClients.Models;
+
+namespace WalletWasabi.WabiSabi.Backend.WebClients;
+
+public class BlockstreamApiClient : ITxPropagationVerifier
+{
+	public BlockstreamApiClient(Network network, HttpClient httpClient)
+	{
+		httpClient.BaseAddress = new Uri(
+			network == Network.TestNet
+				? "https://blockstream.info/testnet/"
+				: "https://blockstream.info/");
```

Comment:
Maybe add another constructor with BaseAddress as a third argument? Would allow to easy add checks to other instances too, as Esplora is open source project, anybody can run an instance of it.